### DISCUSSION TITLE
Scan quality Fujitsu Software vs Sane?

### DISCUSSION THREAD

**Discussion by wpauser-sps** [rocketraman/sane-scan-pdf#49] [General]
Are there any good defaults to achieve a similar scan quality like with the "original" fujitsu software?

- If I switch from LineArt to Color --skip-empty-pages is no longer working
- Setting a Size does not work on the S1300i (only on the S1500) i need --no-default-size otherwise I will get "scanadf: invalid option -- 'y'"
- When scanning without a default size I see the borders

I would appreciate some tips

Thanks for the great script

**Comment by rocketraman** [comment]
I've converted this from an issue to a discussion. If we identify specific problems in the script that need to be resolved, we'll spin those out into separate issues.

> If I switch from LineArt to Color --skip-empty-pages is no longer working

The empty page detection is done by the script using ImageMagick. The current code does not appear to handle color pages very well. When scanning in color, is the paper a different color? Or is the paper still white?

Can you provide an example scan?

@sarmbruster originally wrote the empty page detection logic, perhaps he has some thoughts here as well.

> Setting a Size does not work on the S1300i (only on the S1500) i need --no-default-size otherwise I will get "scanadf: invalid option -- 'y'"

What does `scanadf -L` show, and what does `scanadf -d <S1300i shown from -L> --help` show?

You can set good defaults for your local hardware by using the config file mechanism. See https://github.com/rocketraman/sane-scan-pdf#configuration.

> When scanning without a default size I see the borders

Probably need to see what SANE thinks your hardware's geometry options are. See previous comment.

**Comment by wpauser-sps** [comment]
I can provide a scan any suggestions where to upload it to?

I did use the following script https://gist.github.com/wpauser-sps/6f330a49b12636a0f86cccacee674d57#file-scan-sh

Output in syslog was

Oct 11 21:07:23 scanberry scanbd: /usr/sbin/scanbd: trigger action for scan for device fujitsu:ScanSnap S1500:89817 with script scan.sh
Oct 11 21:07:23 scanberry scanbd[771]: Scanning...
Oct 11 21:07:23 scanberry scanbd[779]: scanadf: rounded value of page-height from 297 to 296.994
Oct 11 21:07:23 scanberry scanbd[779]: scanadf: rounded value of page-width from 210 to 210.01
Oct 11 21:07:23 scanberry scanbd[779]: scanadf: rounded value of br-x from 210 to 210.01
Oct 11 21:07:23 scanberry scanbd[779]: scanadf: rounded value of br-y from 297 to 296.994
Oct 11 21:07:30 scanberry scanbd[779]: Scanned document /tmp/scan.94av8FJ5cA/scan-0001
Oct 11 21:07:30 scanberry scanbd[784]: scan_perpage:
Oct 11 21:07:30 scanberry scanbd[784]: scan_perpage: -------------------------------------------------------------------------------
Oct 11 21:07:30 scanberry scanbd[784]: scan_perpage: Post-processing scanned page /tmp/scan.94av8FJ5cA/scan-0001, deskew=1, searchable=1, skip-empty-pages=1, white-threshold=99.8, brightness-contrast-sw=...
Oct 11 21:07:30 scanberry scanbd[784]: scan_perpage: -------------------------------------------------------------------------------
Oct 11 21:07:33 scanberry scanbd[779]: Scanned document /tmp/scan.94av8FJ5cA/scan-0002
Oct 11 21:07:33 scanberry scanbd[779]: Scanned 2 pages
Oct 11 21:07:35 scanberry scanbd[784]: scan_perpage: /tmp/scan.94av8FJ5cA/scan-0001 has 4.71858 % white
Oct 11 21:07:35 scanberry scanbd[784]: scan_perpage: Applying unpaper post-processing to image data...
Oct 11 21:08:31 scanberry scanbd[801]: unpaper 6.1
Oct 11 21:08:31 scanberry scanbd[801]: License GPLv2: GNU GPL version 2.
Oct 11 21:08:31 scanberry scanbd[801]: This is free software: you are free to change and redistribute it.
Oct 11 21:08:31 scanberry scanbd[801]: There is NO WARRANTY, to the extent permitted by law.
Oct 11 21:08:31 scanberry scanbd[801]: -------------------------------------------------------------------------------
Oct 11 21:08:31 scanberry scanbd[801]: Processing sheet #1: /tmp/scan.94av8FJ5cA/scan-0001 -> /tmp/scan.94av8FJ5cA/unpaper-scan-0001
Oct 11 21:08:31 scanberry scanbd[801]: input-file for sheet 1: /tmp/scan.94av8FJ5cA/scan-0001
Oct 11 21:08:31 scanberry scanbd[801]: output-file for sheet 1: /tmp/scan.94av8FJ5cA/unpaper-scan-0001
Oct 11 21:08:31 scanberry scanbd[801]: sheet size: 2480x3508
Oct 11 21:08:31 scanberry scanbd[801]: ...
Oct 11 21:08:31 scanberry scanbd[801]: noise-filter ... deleted 527 clusters.
Oct 11 21:08:31 scanberry scanbd[801]: blur-filter... deleted 342 pixels.
Oct 11 21:08:31 scanberry scanbd[801]: auto-masking (1240,1754): 0,0,2480,3507 (invalid detection, using full page size)
Oct 11 21:08:31 scanberry scanbd[801]: gray-filter... deleted 22157000 pixels.
Oct 11 21:08:31 scanberry scanbd[801]: auto-masking (1240,1754): 145,0,1830,3507
Oct 11 21:08:31 scanberry scanbd[801]: detected rotation left: [145,0,1830,3507]: -0.008727
Oct 11 21:08:31 scanberry scanbd[801]: detected rotation right: [145,0,1830,3507]: 0.000000
Oct 11 21:08:31 scanberry scanbd[801]: rotation average: -0.004363  deviation: 0.006171  rotation-scan-deviation (maximum): 0.017453  [145,0,1830,3507]
Oct 11 21:08:31 scanberry scanbd[801]: rotate (1240,1754): -0.004363
Oct 11 21:08:31 scanberry scanbd[801]: auto-masking (1240,1754): 145,0,1835,3507
Oct 11 21:08:31 scanberry scanbd[801]: centering mask [145,0,1835,3507] (1240,1754): 250, 0
Oct 11 21:08:31 scanberry scanbd[801]: border detected: (0,195,1,1) in [0,0,2479,3507]
Oct 11 21:08:31 scanberry scanbd[801]: aligning mask [0,195,2478,3506] (0,97): 0, -98
Oct 11 21:08:31 scanberry scanbd[801]: writing output.
Oct 11 21:08:31 scanberry scanbd[802]: [ppm_pipe @ 0x71a670] Stream #0: not enough frames to estimate rate; consider increasing probesize
Oct 11 21:08:31 scanberry scanbd[802]: [image2 @ 0x71df90] Using AVStream.codec to pass codec parameters to muxers is deprecated, use AVStream.codecpar instead.
Oct 11 21:08:31 scanberry scanbd[802]: [image2 @ 0x71df90] Encoder did not produce proper pts, making some up.
Oct 11 21:08:31 scanberry scanbd[802]: real#0110m53.957s
Oct 11 21:08:31 scanberry scanbd[802]: user#0110m51.032s
Oct 11 21:08:31 scanberry scanbd[802]: sys#0110m0.811s
Oct 11 21:08:31 scanberry scanbd[784]: scan_perpage: Converting image data to searchable pdf...
Oct 11 21:08:31 scanberry scanbd[784]: scan_perpage: ...Running convert
Oct 11 21:08:33 scanberry scanbd[827]: real#0110m1.869s
Oct 11 21:08:33 scanberry scanbd[827]: user#0110m0.217s
Oct 11 21:08:33 scanberry scanbd[827]: sys#0110m0.425s
Oct 11 21:08:33 scanberry scanbd[784]: scan_perpage: ...Running tesseract
Oct 11 21:09:09 scanberry scanbd[852]: Tesseract Open Source OCR Engine v4.1.1 with Leptonica
Oct 11 21:09:09 scanberry scanbd[852]: Page 1
Oct 11 21:09:09 scanberry scanbd[852]: real#0110m34.618s
Oct 11 21:09:09 scanberry scanbd[852]: user#0111m17.124s
Oct 11 21:09:09 scanberry scanbd[852]: sys#0110m0.831s
Oct 11 21:09:09 scanberry scanbd[784]: scan_perpage:
Oct 11 21:09:09 scanberry scanbd[784]: scan_perpage: Scan page processing done, status = 0
Oct 11 21:09:09 scanberry scanbd[793]: scan_perpage:
Oct 11 21:09:09 scanberry scanbd[793]: scan_perpage: -------------------------------------------------------------------------------
Oct 11 21:09:09 scanberry scanbd[793]: scan_perpage: Post-processing scanned page /tmp/scan.94av8FJ5cA/scan-0002, deskew=1, searchable=1, skip-empty-pages=1, white-threshold=99.8, brightness-contrast-sw=...
Oct 11 21:09:09 scanberry scanbd[793]: scan_perpage: -------------------------------------------------------------------------------
Oct 11 21:09:13 scanberry scanbd[793]: scan_perpage: /tmp/scan.94av8FJ5cA/scan-0002 has 0.0599321 % white
Oct 11 21:09:13 scanberry scanbd[793]: scan_perpage: Applying unpaper post-processing to image data...
Oct 11 21:10:32 scanberry scanbd[889]: unpaper 6.1
Oct 11 21:10:32 scanberry scanbd[889]: License GPLv2: GNU GPL version 2.
Oct 11 21:10:32 scanberry scanbd[889]: This is free software: you are free to change and redistribute it.
Oct 11 21:10:32 scanberry scanbd[889]: There is NO WARRANTY, to the extent permitted by law.
Oct 11 21:10:32 scanberry scanbd[889]: -------------------------------------------------------------------------------
Oct 11 21:10:32 scanberry scanbd[889]: Processing sheet #1: /tmp/scan.94av8FJ5cA/scan-0002 -> /tmp/scan.94av8FJ5cA/unpaper-scan-0002
Oct 11 21:10:32 scanberry scanbd[889]: input-file for sheet 1: /tmp/scan.94av8FJ5cA/scan-0002
Oct 11 21:10:32 scanberry scanbd[889]: output-file for sheet 1: /tmp/scan.94av8FJ5cA/unpaper-scan-0002
Oct 11 21:10:32 scanberry scanbd[889]: sheet size: 2480x3508
Oct 11 21:10:32 scanberry scanbd[889]: ...
Oct 11 21:10:32 scanberry scanbd[889]: noise-filter ... deleted 2 clusters.
Oct 11 21:10:32 scanberry scanbd[889]: blur-filter... deleted 0 pixels.
Oct 11 21:10:32 scanberry scanbd[889]: auto-masking (1240,1754): 0,0,2480,3507 (invalid detection, using full page size)
Oct 11 21:10:32 scanberry scanbd[889]: gray-filter... deleted 52688500 pixels.
Oct 11 21:10:32 scanberry scanbd[889]: auto-masking (1240,1754): 0,0,2480,3507 (invalid detection, using full page size)
Oct 11 21:10:32 scanberry scanbd[889]: detected rotation left: [0,0,2480,3507]: 0.000000
Oct 11 21:10:32 scanberry scanbd[889]: detected rotation right: [0,0,2480,3507]: 0.000000
Oct 11 21:10:32 scanberry scanbd[889]: rotation average: 0.000000  deviation: 0.000000  rotation-scan-deviation (maximum): 0.017453  [0,0,2480,3507]
Oct 11 21:10:32 scanberry scanbd[889]: rotate (1240,1754): 0.000000
Oct 11 21:10:32 scanberry scanbd[889]: auto-masking (1240,1754): 0,0,2480,3507 (invalid detection, using full page size)
Oct 11 21:10:32 scanberry scanbd[889]: centering mask [0,0,2480,3507] (1240,1754): 0, 0 - NO CENTERING (would shift area outside visible image)
Oct 11 21:10:32 scanberry scanbd[889]: border detected: (0,1200,1,1) in [0,0,2479,3507]
Oct 11 21:10:32 scanberry scanbd[889]: aligning mask [0,1200,2478,3506] (0,600): 0, -600
Oct 11 21:10:32 scanberry scanbd[889]: writing output.
Oct 11 21:10:32 scanberry scanbd[890]: [ppm_pipe @ 0x7bc670] Stream #0: not enough frames to estimate rate; consider increasing probesize
Oct 11 21:10:32 scanberry scanbd[890]: [image2 @ 0x7bfab0] Using AVStream.codec to pass codec parameters to muxers is deprecated, use AVStream.codecpar instead.
Oct 11 21:10:32 scanberry scanbd[890]: [image2 @ 0x7bfab0] Encoder did not produce proper pts, making some up.
Oct 11 21:10:32 scanberry scanbd[890]: real#0111m18.460s
Oct 11 21:10:32 scanberry scanbd[890]: user#0111m16.179s
Oct 11 21:10:32 scanberry scanbd[890]: sys#0110m0.700s
Oct 11 21:10:32 scanberry scanbd[793]: scan_perpage: Converting image data to searchable pdf...
Oct 11 21:10:32 scanberry scanbd[793]: scan_perpage: ...Running convert
Oct 11 21:10:35 scanberry scanbd[916]: real#0110m2.083s
Oct 11 21:10:35 scanberry scanbd[916]: user#0110m0.279s
Oct 11 21:10:35 scanberry scanbd[916]: sys#0110m0.370s
Oct 11 21:10:35 scanberry scanbd[793]: scan_perpage: ...Running tesseract
Oct 11 21:10:41 scanberry scanbd[942]: Tesseract Open Source OCR Engine v4.1.1 with Leptonica
Oct 11 21:10:41 scanberry scanbd[942]: Page 1
Oct 11 21:10:41 scanberry scanbd[942]: Empty page!!
Oct 11 21:10:41 scanberry scanbd[942]: real#0110m5.253s
Oct 11 21:10:41 scanberry scanbd[942]: user#0110m4.815s
Oct 11 21:10:41 scanberry scanbd[942]: sys#0110m0.430s
Oct 11 21:10:41 scanberry scanbd[793]: scan_perpage:
Oct 11 21:10:41 scanberry scanbd[793]: scan_perpage: Scan page processing done, status = 0
Oct 11 21:10:41 scanberry scanbd[771]: Processing 2 pages
Oct 11 21:10:41 scanberry scanbd[771]: Concatenating pdfs...
Oct 11 21:10:42 scanberry scanbd[771]: Done.

**Reply by wpauser-sps** [reply]
File was uploaded to https://paste.c-net.org/HollyMighty - I hope that works

**Reply by rocketraman** [reply]
Looks like you have enabled unpaper. I find unpaper sometimes does some weird things with the scan. Can you do a few runs without `--unpaper`? Also, for experimentation purposes you can try turning off `--deskew`. While we're trying to get the driver options right, its best to do as little post-processing as possible.

Also don't forget to post the output of `scanadf -L --help`.

**Reply by wkocmann** [reply]
Here the same sheet without --unpaper and without --deskew https://paste.c-net.org/HeighUnseemly

**Reply by wkocmann** [reply]
Oddly `scanadf -L` returns nothing so does `scanadf -d`

**Reply by wkocmann** [reply]
It seems that scanadf -L is not working because sane is configured to use net for scanbd to work and not block the device
On another machine without scanbd it is reporting 

```
Options specific to device `fujitsu:ScanSnap S1500:89817':
  Standard:
    --source ADF Front|ADF Back|ADF Duplex [ADF Front]
        Selects the scan source (such as a document-feeder).
    --mode Lineart|Halftone|Gray|Color [Lineart]
        Selects the scan mode (e.g., lineart, monochrome, or color).
    --resolution 50..600dpi (in steps of 1) [600]
        Sets the resolution of the scanned image.
  Geometry:
    --page-width 0..221.121mm (in steps of 0.0211639) [215.872]
        Specifies the width of the media.  Required for automatic centering of
        sheet-fed scans.
    --page-height 0..876.695mm (in steps of 0.0211639) [279.364]
        Specifies the height of the media.
    -l 0..215.872mm (in steps of 0.0211639) [0]
        Top-left x position of scan area.
    -t 0..279.364mm (in steps of 0.0211639) [0]
        Top-left y position of scan area.
    -x 0..215.872mm (in steps of 0.0211639) [215.872]
        Width of scan-area.
    -y 0..279.364mm (in steps of 0.0211639) [279.364]
        Height of scan-area.
  Enhancement:
    --brightness -127..127 (in steps of 1) [0]
        Controls the brightness of the acquired image.
    --contrast -127..127 (in steps of 1) [0]
        Controls the contrast of the acquired image.
    --threshold 0..255 (in steps of 1) [0]
        Select minimum-brightness to get a white point
    --rif[=(yes|no)] [no]
        Reverse image format
    --ht-type Default|Dither|Diffusion [inactive]
        Control type of halftone filter
    --ht-pattern 0..3 (in steps of 1) [inactive]
        Control pattern of halftone filter
    --emphasis -128..127 (in steps of 1) [0]
        Negative to smooth or positive to sharpen image
    --variance 0..255 (in steps of 1) [0]
        Set SDTC variance rate (sensitivity), 0 equals 127
  Advanced:
    --ald[=(yes|no)] [no]
        Scanner detects paper lower edge. May confuse some frontends.
    --df-action Default|Continue|Stop [Default]
        Action following double feed error
    --df-skew[=(yes|no)] [inactive]
        Enable double feed error due to skew
    --df-thickness[=(yes|no)] [inactive]
        Enable double feed error due to paper thickness
    --df-length[=(yes|no)] [inactive]
        Enable double feed error due to paper length
    --df-diff Default|10mm|15mm|20mm [inactive]
        Difference in page length to trigger double feed error
    --dropoutcolor Default|Red|Green|Blue [Default]
        One-pass scanners use only one color during gray or binary scanning,
        useful for colored paper or ink
    --buffermode Default|Off|On [Off]
        Request scanner to read pages quickly from ADF into internal memory
    --sleeptimer 0..60 (in steps of 1) [0]
        Time in minutes until the internal power supply switches to sleep mode
    --lowmemory[=(yes|no)] [no]
        Limit driver memory usage for use in embedded systems. Causes some
        duplex transfers to alternate sides on each call to sane_read. Value of
        option 'side' can be used to determine correct image. This option
        should only be used with custom front-end software.
    --swdeskew[=(yes|no)] [no]
        Request driver to rotate skewed pages digitally.
    --swdespeck 0..9 (in steps of 1) [0]
        Maximum diameter of lone dots to remove from scan.
    --swcrop[=(yes|no)] [no]
        Request driver to remove border from pages digitally.
    --swskip 0..100% (in steps of 0.100006) [0]
        Request driver to discard pages with low percentage of dark pixels
  Sensors:

```

**Reply by rocketraman** [reply]
Those are the options for the S1500. Any chance you can get that output for the other scanner?

**Reply by wkocmann** [reply]
only by the end of the week
